src/coreclr/jit/compiler.cpp

```diff
@@ -560,6 +560,23 @@ bool Compiler::isTrivialPointerSizedStruct(CORINFO_CLASS_HANDLE clsHnd) const
 }
 #endif // TARGET_X86
 
+bool Compiler::isNativePrimitiveStructType(CORINFO_CLASS_HANDLE clsHnd)
```

Comment:
Just to confirm, adding a member to this struct basically means it will be passed as the underlying `_value` rather than as a struct type itself, correct?

So if we add additional ABI types in the future, such as `Int128`, we might want to add them here?

Reply:
Effectively that is what happens.  This check opts the type out of the "is this a struct" disqualifications and uses the traditional calling convention rules (which don't differentiate between a "struct of an int" and an "int"). I thought about doing this in a way that actually unwrapped it, but that would have also affected the managed calling conventions and would have likely been implemented on the VM side instead of the JIT side.

Reply:
> This check opts the type out of the "is this a struct" disqualifications and uses the traditional calling convention rules (which don't differentiate between a "struct of an int" and an "int")

How does that impact `thiscall` where this is differentiated (i.e. where `typedef int HRESULT;` and `struct HRESULT { int value; }` are different)?

Reply:
A `CLong` return type on a `thiscall` method or call will act as a C++ `long` return type, not as a `struct CLong { long value; };` type. This is true for all 3 types.

Reply:
👍, just wanted to confirm since you said "uses the traditional calling convention rules"